src/player.cpp

```diff
@@ -512,17 +514,17 @@ void player::action_taken()
 void player::update_morale()
 {
     // Decay existing morale entries.
-    for( size_t i = 0; i < morale.size(); i++ ) {
+    for( size_t i = morale.size() - 1; i > 0; i-- ) {
```

Comment:
OK, this is the part that prevents morale decay.
The condition `i > 0` means that the loop body will not be called for `i == 0` and so `morale[0].proceed` will never be called.